tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/cos/CosFileSegment.java

```diff
@@ -0,0 +1,250 @@
+        ByteBuffer byteBuffer = ByteBuffer.allocate(length);
+        try {
+            GetObjectRequest getObjectRequest = new GetObjectRequest(cosBucketName, filepath);
+            COSObject cosObject = cosClient.getObject(getObjectRequest);
+            int readed = cosObject.getObjectContent().read(byteBuffer.array(), (int) position, length);
+            if (readed != length) {
+                throw new IOException("read from cos failed,length is not matched");
+            }
+            byteBuffer.flip();
+
+            attributesBuilder.put(LABEL_SUCCESS, true);
+            long costTime = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+            TieredStoreMetricsManager.providerRpcLatency.record(costTime, attributesBuilder.build());
+
+            Attributes metricsAttributes = newAttributesBuilder()
+                .put(LABEL_OPERATION, OPERATION_POSIX_READ)
+                .build();
+            int downloadedBytes = byteBuffer.remaining();
+            TieredStoreMetricsManager.downloadBytes.record(downloadedBytes, metricsAttributes);
+
+            future.complete(byteBuffer);
+        } catch (IOException e) {
+            long costTime = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+            attributesBuilder.put(LABEL_SUCCESS, false);
+            TieredStoreMetricsManager.providerRpcLatency.record(costTime, attributesBuilder.build());
+            logger.error("CosFileSegment#read0: read file {} failed: position: {}, length: {}",
+                filepath, position, length, e);
+            future.completeExceptionally(e);
+        }
+        return future;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> commit0(TieredFileSegmentInputStream inputStream, long position, int length,
+        boolean append) {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        AttributesBuilder attributesBuilder = newAttributesBuilder()
+            .put(LABEL_OPERATION, OPERATION_POSIX_WRITE);
+
+        CompletableFuture<Boolean> future = new CompletableFuture<>();
+        try {
+            TieredStoreExecutor.commitExecutor.execute(() -> {
+                try {
+                    byte[] byteArray = ByteStreams.toByteArray(inputStream);
+                    if (byteArray.length != length) {
+                        logger.error("CosFileSegment#commit0: append file {} failed: real data size: {}, is not equal to length: {}",
+                            filepath, byteArray.length, length);
+                        future.complete(false);
+                        return;
+                    }
+
+                    ByteBuf buffer = Unpooled.copiedBuffer(byteArray);
+                    ByteBufInputStream backedInputStream = new ByteBufInputStream(buffer);
+                    AppendObjectRequest appendObjectRequest = new AppendObjectRequest(this.cosBucketName, filepath, backedInputStream);
+                    appendObjectRequest.setPosition(position);
```

Comment:
I think it should use TieredStoreException here to tell the real position if append position is not equal to the file size.